Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -355,6 +355,125 @@ void RewriteGlobalVariables::visitEntryPoint(AST::Function& function, AST::Stage
     insertLocalDefinitions(function, usedGlobals.privateGlobals);
 }
 
+static BindGroupLayoutEntry::BindingMember bindingMemberForGlobal(auto& global)
+{
+    auto* variable = global.declaration;
+    ASSERT(variable);
+    auto* type = variable->storeType();
+    ASSERT(type);
+
+    using namespace WGSL::Types;
+    return WTF::switchOn(*type, [&](const Primitive& primitive) -> BindGroupLayoutEntry::BindingMember {
+        switch (primitive.kind) {
+        case Types::Primitive::AbstractInt:
+        case Types::Primitive::I32:
+        case Types::Primitive::U32:
+        case Types::Primitive::AbstractFloat:
+        case Types::Primitive::F32:
+        case Types::Primitive::Void:
+        case Types::Primitive::Bool:
+            RELEASE_ASSERT_NOT_REACHED();
```

Comment:
Really? Seems crazy that `vec4<f32>` would work, but `f32` wouldn't work

Reply:
I did not test f32, thank you for noticing, I will try that and update the patch.